src/input/mod.rs

```diff
@@ -0,0 +1,142 @@
+use std::fs::File;
+use std::{fmt, io};
+use std::error::Error;
+use std::io::{BufRead, BufReader, IsTerminal, Read};
+use std::path::{Path, PathBuf};
+use ::base64::DecodeError;
+use ::hex::FromHexError;
+use clap::Parser;
+use crate::input::base64::{base64_decode, is_valid_base64};
+use crate::input::hex::{hex_decode, is_valid_hex};
+
+pub mod base64;
+pub mod hex;
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Cli {
+    #[arg(short = 'f', long, value_name = "FILE")]

```

Comment:
add about attribute to struct derive
https://docs.rs/clap/latest/clap/_derive/index.html#command-attributes